src/i2c.rs

```diff
@@ -220,16 +220,16 @@ macro_rules! wait_for_flag {
         let sr1 = $i2c.sr1.read();
 
         if sr1.berr().bit_is_set() {
-            $i2c.sr1.write(|w| w.berr().clear_bit());
+            $i2c.sr1.modify(|_r, w| w.berr().clear_bit());
```

Comment:
```suggestion
            $i2c.sr1.write(|w| w
                 .bits(sr1.bits())
                 .berr().clear_bit()
            );
```
Something like this

Reply:
In theory, one should use an all-one mask instead of the previous register value, because the register value can be changed during the read-modify-write sequence and this code will end up clearing bits it wasn't supposed to clear.

Reply:
I agree
![изображение](https://user-images.githubusercontent.com/3072754/103154244-2b824400-47a7-11eb-8e9e-1ca796cbf5d1.png)